rest_framework_json_api/renderers.py

```diff
@@ -273,6 +273,30 @@ def extract_relationships(cls, fields, resource, resource_instance):
         return utils.format_keys(data)
 
     @classmethod
+    def extract_relation_instance(cls, field_name, field, resource_instance, serializer):
```

Comment:
I have added a docstring.

Considering the levels of exceptions I also think this is certainly not ideal but I cannot think of an alternative way to do it though.

For example calling `getattr` with default argument `None` wouldn't work, as an attribute could actually be `None`. Calling `hasattr` wouldn't make things better either as internally it also does a try-except (https://docs.python.org/3/library/functions.html#hasattr).
It could be argued whether the serializer method case is really needed but otherwise I guess we will need to live with it.